src/coinbase/staking_balance.ts

```diff
@@ -0,0 +1,120 @@
+import { StakingBalance as StakingBalanceModel } from "../client";
+import { Balance } from "./balance";
+import { Coinbase } from "./coinbase";
+
+/**
+ * A representation of a staking balance earned on a network for a given asset.
+ */
+export class StakingBalance {
+  private model: StakingBalanceModel;
+
+  /**
+   * Creates the StakingBalance object.
+   *
+   * @param model - The underlying staking balance object.
+   */
+  constructor(model: StakingBalanceModel) {
+    this.model = model;
+  }
+
+  /**
+   * Returns a list of StakingBalances for the provided network, asset, and address.
+   *
+   * @param networkId - The network ID.
+   * @param assetId - The asset ID.
+   * @param address - The address ID.
+   * @param startTime - The start time.
+   * @param endTime - The end time.
+   * @returns The staking balances.
+   */
+  public static async list(
```

Comment:
Should this be a method on the `Address` class [ref](https://github.com/coinbase/coinbase-sdk-nodejs/blob/7ca44cb7c4cdaa02c43d03eab39d8fb5dbc83e5f/src/coinbase/address.ts#L15) instead of static method?

Reply:
we have an equivalent for [listing staking rewards](https://github.com/coinbase/coinbase-sdk-nodejs/blob/7ca44cb7c4cdaa02c43d03eab39d8fb5dbc83e5f/src/coinbase/staking_reward.ts#L39) , [I think we need both](https://github.com/coinbase/coinbase-sdk-nodejs/blob/7ca44cb7c4cdaa02c43d03eab39d8fb5dbc83e5f/src/coinbase/address.ts#L91)

Reply:
I see. That pattern is fine. Can you please add the `Address` class change to this PR.

Reply:
added, wondering if I can test the change to Address, currently only unit tests added to all related files including `Wallet`, `External Address` & `Wallet Address`

Reply:
Cool. Thanks! You can also call this method on an `ExternalAddress` subclass instance if you want to test on a specific address.

Reply:
thanks, that works as well
```
const addr = new ExternalAddress(Coinbase.networks.EthereumMainnet,  "0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a")

const addr = new ExternalAddress(Coinbase.networks.EthereumMainnet,  "0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a")
addr.historicalStakingBalances(Coinbase.assets.Eth, "2024-07-01T00:00:00Z", "2024-07-03T00:00:00Z").then(balances => {
    balances.forEach(balance => {console.log(balance)});
} ).catch(error => {
    console.log(error);
});

StakingBalance {
  model: {
    address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a',
    bonded_stake: { amount: '32', asset: [Object] },
    date: '2024-07-02',
    participant_type: 'VALIDATOR',
    unbonded_balance: { amount: '5.479839548', asset: [Object] }
  }
}
```